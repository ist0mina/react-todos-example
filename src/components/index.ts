export * from './nav-button';
export * from './animation';
export * from './custom-select';
export * from './custom-datetime';