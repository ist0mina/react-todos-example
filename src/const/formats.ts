export const STANDART_DATE_FORMAT = 'DD.MM.YYYY';
export const STANDART_DATE_TIME_FORMAT = 'DD.MM.YYYY HH:mm';