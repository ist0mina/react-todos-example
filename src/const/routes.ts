export enum RoutePaths {
    ROOT = '/',
    TODO_DETAIL = '/todo'
}