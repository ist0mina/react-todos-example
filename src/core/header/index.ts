export * from './Header';
export * from './HeaderContainer';
export * from './types';