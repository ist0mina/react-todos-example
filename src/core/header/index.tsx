import React from 'react';
import { Jumbotron } from 'react-bootstrap';

const Header = () => {
    return (
        <Jumbotron>
            <h1 className="text-center">Todo list</h1>
        </Jumbotron>
    );
};

export default Header;