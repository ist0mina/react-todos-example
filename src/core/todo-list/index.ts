import TodoListContainer from './TodoListContainer';

export * from './types';

export default TodoListContainer;