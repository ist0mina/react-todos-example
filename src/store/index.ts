import { createStore, applyMiddleware } from 'redux';
import { createBrowserHistory } from 'history';
import createSagaMiddleware from 'redux-saga';
import { composeWithDevTools } from 'redux-devtools-extension';
import rootSaga from './rootSaga';
import rootReducer from './rootReducer';
import { routerMiddleware } from 'connected-react-router';

export const history = createBrowserHistory();

const sagaMiddlware = createSagaMiddleware();

const composeEnhacers = composeWithDevTools({});

export const createStoreWithMiddleware = (reducer: any, initialState = {}) => 
    createStore(reducer, initialState, composeEnhacers(applyMiddleware(routerMiddleware(history), sagaMiddlware)));

const store = createStoreWithMiddleware(rootReducer(history));

if (process.env.NODE_ENV !== 'test'){
    sagaMiddlware.run(rootSaga);
}

export default store;