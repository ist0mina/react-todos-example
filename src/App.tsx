import React from 'react';
import { Provider } from 'react-redux';
import store from './store';
import './App.css';
import Layout from './core/layout';
import TodoPage from './core/todo-page';

const App: React.FC = () => (
  <Provider store={store}>    
    <Layout>
      <TodoPage />
    </Layout>    
  </Provider>
);

export default App;
