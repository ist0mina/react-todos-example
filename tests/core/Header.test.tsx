import { makeMountrender, reduxify, snapshotify, mockStore } from '../utils';

import { HeaderContainer } from '../../src/core/header';
import { Routes } from '../../src/core/routes';
import { TodoList } from '../../src/core/todo-list';
import { TodoDetail } from '../../src/core/todo-detail';

import { history } from '../../src/store';

import { RoutePaths } from '../../src/const/routes';

describe('<HeaderContainer>', () => {
    
    describe('active buttons', () => {
        let wrapper: any;

        beforeEach(() => {
            const component = reduxify({ Component: HeaderContainer });
            wrapper = makeMountrender(component)();
        }); 

        it('active todo list button', () => {      
            history.push(RoutePaths.ROOT);
            expect(snapshotify(wrapper)).toMatchSnapshot();
        });    

        it('active todo detail button', () => {
            history.push(RoutePaths.TODO_DETAIL);
            expect(snapshotify(wrapper)).toMatchSnapshot();
        });
    });    

    describe('simulate', () => {
        it('simulate link click', () => {
            const store = mockStore();

            const component = reduxify({ Component: Routes, store });
            const wrapper = makeMountrender(component)();            

            wrapper.find(`a[href="${RoutePaths.ROOT}"]`).simulate('click', { button: 0 });
            expect(wrapper.find(TodoList)).toHaveLength(1);

            wrapper.find(`a[href="${RoutePaths.TODO_DETAIL}"]`).simulate('click', { button: 0 });
            expect(wrapper.find(TodoDetail)).toHaveLength(1);
        });        
    });    
});