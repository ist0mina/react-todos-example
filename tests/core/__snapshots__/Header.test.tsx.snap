// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderContainer> active buttons active todo detail button 1`] = `"<header><nav class=\\"navbar navbar-expand navbar-light bg-light\\"><div class=\\"mr-auto navbar-nav\\"><div class=\\"nav-item\\"><a href=\\"/\\" class=\\"nav-link\\"><button type=\\"button\\" class=\\"btn btn-outline-secondary\\">Todo list</button></a></div><div class=\\"nav-item\\"><a href=\\"/todo\\" class=\\"nav-link\\"><button type=\\"button\\" class=\\"nav-active btn btn-outline-secondary\\">Todo detail</button></a></div></div></nav></header>"`;

exports[`<HeaderContainer> active buttons active todo list button 1`] = `"<header><nav class=\\"navbar navbar-expand navbar-light bg-light\\"><div class=\\"mr-auto navbar-nav\\"><div class=\\"nav-item\\"><a href=\\"/\\" class=\\"nav-link\\"><button type=\\"button\\" class=\\"nav-active btn btn-outline-secondary\\">Todo list</button></a></div><div class=\\"nav-item\\"><a href=\\"/todo\\" class=\\"nav-link\\"><button type=\\"button\\" class=\\"btn btn-outline-secondary\\">Todo detail</button></a></div></div></nav></header>"`;
